modules/apps/27-interchain-accounts/host/keeper/relay_test.go

```diff
@@ -451,35 +455,378 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
+					0,
+				)
+
+				txResponse, err := suite.chainB.GetSimApp().ICAHostKeeper.OnRecvPacket(suite.chainB.GetContext(), packet)
+
+				if tc.expPass {
+					suite.Require().NoError(err)
+					suite.Require().NotNil(txResponse)
+				} else {
+					suite.Require().Error(err)
+					suite.Require().Nil(txResponse)
+				}
+			})
+		}
+	}
+}
+
+func (suite *KeeperTestSuite) TestJSONOnRecvPacket() {
+	var (
+		path       *ibctesting.Path
+		packetData []byte
+	)
+	interchainAccountAddr := "cosmos15ulrf36d4wdtrtqzkgaan9ylwuhs7k7qz753uk"
+
+	testCases := []struct {
+		msg      string
+		malleate func(icaAddress string)
+		expPass  bool
+	}{
+		{
+			"interchain account successfully executes an arbitrary message type using the * (allow all message types) param",
+			func(icaAddress string) {
+				// Populate the gov keeper in advance with an active proposal
+				testProposal := &govtypes.TextProposal{
+					Title:       "IBC Gov Proposal",
+					Description: "tokens for all!",
+				}
+
+				proposalMsg, err := govv1.NewLegacyContent(testProposal, interchainAccountAddr)
+				suite.Require().NoError(err)
+
+				proposal, err := govv1.NewProposal([]sdk.Msg{proposalMsg}, govtypes.DefaultStartingProposalID, suite.chainA.GetContext().BlockTime(), suite.chainA.GetContext().BlockTime(), "test proposal", "title", "Description", sdk.AccAddress(interchainAccountAddr))
+				suite.Require().NoError(err)
+
+				suite.chainB.GetSimApp().GovKeeper.SetProposal(suite.chainB.GetContext(), proposal)
+				suite.chainB.GetSimApp().GovKeeper.ActivateVotingPeriod(suite.chainB.GetContext(), proposal)
+
+				msgBytes := []byte(`{
+					"messages": [
+						{
+							"@type": "/cosmos.gov.v1beta1.MsgVote",
+							"voter": "` + icaAddress + `",
+							"proposal_id": 1,
+							"option": 1
+						}
+					]
+				}`)
+				// this is the way cosmwasm encodes byte arrays by default
+				// golang doesn't use this encoding by default, but it can still deserialize:
+				byteArrayString := strings.Join(strings.Fields(fmt.Sprint(msgBytes)), ",")
+
+				packetData = []byte(`{
+					"type": 1,
+					"data":` + byteArrayString + `
+				}`)
+
+				params := types.NewParams(true, []string{"*"})
+				suite.chainB.GetSimApp().ICAHostKeeper.SetParams(suite.chainB.GetContext(), params)
+			},
+			true,
+		},
+		{
+			"interchain account successfully executes banktypes.MsgSend",
+			func(icaAddress string) {
+				msgBytes := []byte(`{
+					"messages": [
+						{
+							"@type": "/cosmos.bank.v1beta1.MsgSend",
+							"from_address": "` + icaAddress + `",
+							"to_address": "cosmos17dtl0mjt3t77kpuhg2edqzjpszulwhgzuj9ljs",
+							"amount": [{ "denom": "stake", "amount": "100" }]
+						}
+					]
+				}`)
+				byteArrayString := strings.Join(strings.Fields(fmt.Sprint(msgBytes)), ",")
+
+				packetData = []byte(`{
+					"type": 1,
+					"data":` + byteArrayString + `
+				}`)
+
+				params := types.NewParams(true, []string{sdk.MsgTypeURL((*banktypes.MsgSend)(nil))})
```

Comment:
can this just be

`params := types.NewParams(true, []string{sdk.MsgTypeURL(nil))`?

Reply:
No, because we are trying to extract the url of `MsgSend`

Reply:
ah right makes sense 👍